### PR TITLE
fix: LoadBase Selector cannot follow the region distribution rules

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

fix LoadBase Selector cannot follow the region distribution rules

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/meta-srv/src/selector/load_based.rs [hunk 1]
```diff
@@ -43,22 +47,44 @@ impl Selector for LoadBasedSelector {
...
+                .await
+                .context(error::TableMetadataManagerSnafu)?;
+
+            if let Some(table_name) = table_name {
+                let route = table_metadata_manager
+                    .table_route_manager()
+                    .get(table_name.table_id())
+                    .await
+                    .context(error::TableMetadataManagerSnafu)?
+                    .with_context(|| error::TableRouteNotFoundSnafu {
```
Comment (MichaelScofield) [inline src/meta-srv/src/selector/load_based.rs:68]: Is it possible that table route value is legitimate to be not found here? For example, the table is dropped while executing this selector.

Comment (WenyXu) [reply src/meta-srv/src/selector/load_based.rs:68]: No, the Drop table will remove all metadata in a single txn.

Comment (MichaelScofield) [reply src/meta-srv/src/selector/load_based.rs:68]: But it's possible the table route value is not there when you retrieve it here. After all, you are not fetching the table name key and table route key at the same time.

**Comment by WenyXu** [comment]
@MichaelScofield @fengjiachun PTAL

**Review by MichaelScofield — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Review by fengys1996 — APPROVED** [review]
LGTM

**Pull request merged**